docs/design-documents/features/storage/TDBStore/TDBStore_design.md

```diff
@@ -0,0 +1,529 @@
+    + [Important data structures](#important-data-structures)
+    + [Initialization and reset](#initialization-and-reset)
+    + [Core APIs](#core-apis)
+    + [Incremental set APIs](#incremental-set-apis)
+    + [Key iterator APIs](#key-iterator-apis)
+- [Usage scenarios and examples](#usage-scenarios-and-examples)
+    + [Standard usage of the class](#standard-usage-of-the-class)
+- [Other information](#other-information)
+    + [Open issues](#open-issues)
+
+
+### Revision history
+
+| Revision 	| Date           	| Authors                                                	| Mbed OS version 	| Comments         	|
+|----------	|----------------	|--------------------------------------------------------	|-----------------	|------------------	|
+| 1.0      	| 16 September 2018	| David Saada ([@davidsaada](https://github.com/davidsaada/)) 	| 5.11+           	| Initial revision 	|
+
+# Introduction
+
+### Overview and background
+
+Tiny Database Storage (TDBStore) is a lightweight module that stores data on flash storage. It is part of of the [KVStore](../KVStore/KVStore_design.md) class family, meaning it supports the get/set interface. It is designed to optimize performance (speed of access), reduce wearing of the flash and minimize storage overhead. It is also resilient to power failures.
+
+### Requirements and assumptions
+
+TDBStore assumes the underlying block device is fully dedicated to it (starting offset 0). If you want to dedicate only a part of the device to TDBStore, use a sliced block device, typically with `SlicingBlockDevice`.
+
+In addition, this feature requires a flash-based block device, such as `FlashIAPBlockDevice` or `SpifBlockDevice`. It can work on top of block devices that don't need erasing before writes, such as `HeapBlockDevice` or `SDBlockDevice`, but requires a flash simulator layer for this purpose, such as the one `FlashSimBlockDevice` offers. 
+
+# System architecture and high-level design
+
+## Design basics
+
+TDBStore includes the following design basics:
+
+- Sequential writes: All writes occur sequentially on the physical storage as records, superseding the previous ones for the same key.  
+- Memory layout areas: The physical storage is divided equally into two areas - active and standby. All writes are made to the end of the active area's free space. When the active area is exhausted, a garbage collection process is invoked, copying only the up-to-date values of all keys to the standby area and making it active.
+- RAM table: Indexes all keys in RAM, allowing fast access to the records in the physical storage.   
+
+### Sequential writes
+
+All writes occur sequentially on the physical storage as records, superseding the previous ones for the same key. Each data record is written right after the last written one. If a key is updated, a new record with this key is written, overriding the previous value of this key. If a key is deleted, a new record with a "deleted" flag is added.
```

Comment:
A lot of this document is in passive voice, and I'm not sure what's doing the action. What's writing data records? What's updating keys? What's deleting keys? Is it you, the user? Please clarify.

Reply:
Thanks. Rephrased these lines.

Reply:
Your changes look good. I came across a couple design docs I missed in my first round of edits, so I'm finishing those up, and then I'll be finished.